### PR TITLE
Use Enumerable.Repeat and code cleanup

### LINKED ISSUE #5080: Consider if it is appropriate to use Enumerable.Repeat to implement DefaultLowerBounds function in src/Compilers/CSharp/Portable/Emitter/Model/ArrayTypeSymbolAdapter.cs
In code review for PR #5060 @gafter asked if we should use ```Enumerable.Repeat``` to implement DefaultLowerBounds, which is currently implemented as follows:
```
private static IEnumerable<int> DefaultLowerBounds(int rank)
{
    for (int i = 0; i < rank; ++i)
        yield return 0;
}
```

### PR DESCRIPTION
**Pull request by leppie**
I try again :)

Fixes #5080 

Did some code cleanup too with C#6 goodies :)

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/roslyn-compiler

File: src/Compilers/CSharp/Portable/Emitter/Model/ArrayTypeSymbolAdapter.cs [hunk 1]
```diff
@@ -43,7 +44,7 @@ IEnumerable<int> Cci.IArrayTypeReference.LowerBounds
 
                 if (lowerBounds.IsDefault)
                 {
-                    return DefaultLowerBounds(this.Rank);
+                    return Enumerable.Repeat(0, Rank);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Emitter/Model/ArrayTypeSymbolAdapter.cs]: There is probably similar code in VB implementation. If we are changing C#, we should change VB as well.

**Comment by davkean** [comment]
@dotnet-bot test eta please

File: src/Compilers/CSharp/Portable/Emitter/Model/ArrayTypeSymbolAdapter.cs (hunk 1 above)
Comment (leppie) [inline src/Compilers/CSharp/Portable/Emitter/Model/ArrayTypeSymbolAdapter.cs]: Not seeing this in VB, but my VB scanning skills are not worth mentioning.

Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Emitter/Model/ArrayTypeSymbolAdapter.cs]: You are correct, it looks like it is already using Repeat.

**Comment by AlekseyTs** [comment]
LGTM

**Comment by VSadov** [comment]
LGTM

**Pull request merged**